src/levanter/inference/service.py

```diff
@@ -36,9 +38,23 @@ class Request:
     n_generations: int
 
 
+@dataclasses.dataclass
+class DecodeResult:
+    """Holds per-(request, choice) decode outputs and status."""
+
+    id: int
+    choice: int
+    token_list: list[int]
+    # Count of newly appended tokens (includes prompt tokens as extracted)
+    tokens_decoded: int = 0
+    done: bool = False
+    # Optional per-token logprobs (not populated yet)
+    logprobs: list[float] | None = None
+
+
 @dataclass(frozen=True)
-class GenerationServiceConfig:
-    """Configuration for GenerationService memory/layout knobs.
+class EngineConfig:
```

Comment:
Is this common terminology? Seems generic vs InferenceEngine, InferenceManager etc.

Reply:
It's pretty common. Vllm uses LlmEngine, but sglang is just Engine https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/entrypoints/engine.py#L92 I dunno. I can do InferenceEngine

Reply:
Up to you, it seemed a bit generic to me in the context of Levanter, but not a big deal.